app/controllers/articles_controller.rb

```diff
@@ -16,8 +16,8 @@ def index
   end
 
   def show
-    @article.count_visit
-    respond_with_article_or_redirect
+    respond_with_article_or_redirect_or_new
+    @article.count_visit if @article.present?
```

Comment:
Not in love with this but it's the only way I know to handle the fact that Draper decorators aren't falsey when they decorate nil objects.